kernel/src/snapshot.rs

```diff
@@ -385,6 +404,97 @@ impl Snapshot {
 
         domain_metadata_configuration(self.log_segment(), domain, engine)
     }
+
+    /// Get the timestamp of this snapshot.
+    ///
+    /// If In-Commit Timestamps (ICT) is enabled for the table, this returns the inCommitTimestamp
+    /// from the CommitInfo action of the commit that created this snapshot.
+    ///
+    /// If ICT is not enabled, this returns the file modification time of the commit file.
+    ///
+    /// This method follows the same logic as the Java Kernel implementation.
+    pub fn get_timestamp(&self, engine: &dyn Engine) -> DeltaResult<i64> {
+        if self.table_configuration().is_in_commit_timestamps_enabled() {
+            // Check if we have a cached value first
+            if let Some(cached_timestamp) = self.cached_timestamp.get() {
+                return self.get_timestamp_from_ict(*cached_timestamp);
+            }
+
+            // Compute and cache the timestamp
+            let computed_timestamp = self.compute_ict_timestamp(engine)?;
+            let _ = self.cached_timestamp.set(computed_timestamp);
+
+            self.get_timestamp_from_ict(computed_timestamp)
+        } else {
+            self.get_file_modification_timestamp()
+        }
```

Comment:
get_timestamp is only ever used to retrieve the in-commit timestamp. I don't think it would be correct to return the file_modification timestamp.